src/rust/ide/src/view/node_editor.rs

```diff
@@ -699,26 +710,52 @@ impl GraphEditorIntegratedWithControllerModel {
         Ok(())
     }
 
+    fn expression_entered_in_ui
+    (&self, local_call:&Option<LocalCall>) -> FallibleResult<()> {
+        if let Some(local_call) = local_call {
+            let method_pointer = local_call.definition.clone();
+            let expression_id  = local_call.call;
+            let controller     = self.controller.clone_ref();
+            let logger         = self.logger.clone_ref();
+            let graph_editor   = self.editor.clone_ref();
+            let enter_action   = async move {
+                match controller.enter_method_pointer(expression_id,&method_pointer).await {
+                    Ok(_) => {
+                        info!(logger,"Entering node.");
+                        let local_call = LocalCall{definition:method_pointer,call:expression_id};
+                        graph_editor.breadcrumbs.frp.push_breadcrumb.emit(&Some(local_call));
+                    },
+                    Err(e) => error!(logger,"Couldn't enter node: {e}")
```

Comment:
This does not belong here. 
Basically all these "sth_in_ui" should only request action from controller but not apply any consequences of controller realizing this action. 
This code belongs to `on_node_entered`. There are code paths allowing node being entered without the ui action here, the breadcrumbs should always stay in sync with controller, not only when change is made from ui.

Reply:
to fix this you should like add method pointer information to the `Notification::EnteredNode` and adjust code around it accordingly

Reply:
@mwu-tow I just provided a change for it. Can you check if it's correct?